stock_picking_force_assign/models/stock_picking.py

```diff
@@ -40,7 +46,7 @@ def _force_assign_find_moves(self):
             precision_digits=result._fields['product_qty'].digits,
         )
 
-        for move in self.mapped('move_lines'):
+        for move in self.mapped('move_lines').filtered(lambda move: move.state != 'cancel'):
```

Comment:
better `continue` within the loop in this case, otherwise you loop through the lines twice